server/src/hardware/output/map/outputmap.cpp

```diff
@@ -586,18 +615,37 @@ void OutputMap::addOutput(OutputChannel ch, uint32_t id)
 void OutputMap::addOutput(OutputChannel ch, uint32_t id, OutputController& outputController)
 {
   m_outputs.emplace_back(getOutput(ch, id, outputController));
-  assert(m_outputs.back());
+  assert(m_outputs.back().first);
 }
 
-std::shared_ptr<Output> OutputMap::getOutput(OutputChannel ch, uint32_t id, OutputController& outputController)
+OutputMap::OutputConnectionPair OutputMap::getOutput(OutputChannel ch, uint32_t id, OutputController& outputController)
 {
   auto output = outputController.getOutput(ch, id, parent());
-  // TODO: connect output value changed
-  return output;
+
+  boost::signals2::connection conn;
+
+  if(auto *single = dynamic_cast<SingleOutput*>(output.get()))
+  {
+    conn = single->onValueChanged.connect([this](bool, const std::shared_ptr<SingleOutput>&){ updateStateFromOutput(); });
+  }
+  else if(auto* pair = dynamic_cast<PairOutput*>(output.get()))
+  {
+    conn = pair->onValueChanged.connect([this](OutputPairValue, const std::shared_ptr<PairOutput>&){ updateStateFromOutput(); });
+  }
+  else if(auto* aspect = dynamic_cast<AspectOutput*>(output.get()))
+  {
+    conn = aspect->onValueChanged.connect([this](int16_t, const std::shared_ptr<AspectOutput>&){ updateStateFromOutput(); });
+  }
+  else if(auto* ecosState = dynamic_cast<ECoSStateOutput*>(output.get()))
+  {
+    conn = ecosState->onValueChanged.connect([this](uint8_t, const std::shared_ptr<ECoSStateOutput>&){ updateStateFromOutput(); });
+  }
```

Comment:
We should really get rid of this, we should add a valueChanged (without value) signal to `Output`, then that can be used to trigger `updateStateFromOutput();`

Reply:
Done! Do you like the new implementation?